packages/admin/cms-admin/src/dam/DataGrid/footer/SelectionFooter.tsx

```diff
@@ -146,3 +213,60 @@ export const DamSelectionFooter: React.VoidFunctionComponent<DamSelectionFooterP
         </>
     );
 };
+
+const StyledCometAdminIconButton = styled(CometAdminIconButton)`
+    color: ${({ theme }) => theme.palette.grey.A100};
+    padding-left: 4px;
+    padding-right: 4px;
+`;
+
+interface IconButtonProps {
+    title: NonNullable<React.ReactNode>;
+    onClick?: () => void;
+    executeMutation?: () => Promise<{ errors: readonly GraphQLError[] | undefined } | undefined>;
+    icon: React.ReactNode;
+    loading?: boolean;
+    hasErrors?: boolean;
+}
+
+const IconButton = ({ title, onClick, executeMutation, icon, loading: externalLoading, hasErrors: externalHasErrors }: IconButtonProps) => {
+    const apolloClient = useApolloClient();
+
+    const [internalLoading, setInternalLoading] = React.useState<boolean>(false);
+    const [internalHasErrors, setInternalHasErrors] = React.useState<boolean>(false);
+
+    const loading = externalLoading ?? internalLoading;
+    const hasErrors = externalHasErrors ?? internalHasErrors;
+
+    const handleClick = async () => {
+        if (executeMutation === undefined) {
+            throw new Error("IconButton: You must either set onClick or executeMutation");
+        }
+
+        setInternalLoading(true);
+
+        const result = await executeMutation();
+
+        if (result) {
+            if (result.errors) {
+                setInternalHasErrors(true);
+                setTimeout(() => {
+                    setInternalHasErrors(false);
+                }, 3000);
+            } else {
+                await apolloClient.refetchQueries({ include: [namedOperations.Query.DamItemsList] });
+                clearDamItemCache(apolloClient.cache);
```

Comment:
You first refetch queries and then clear the cache? Wouldn't this throw away the fetched queries?

Reply:
Yes, you are right. The refetch is not necessary